src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.tsx

```diff
@@ -28,49 +35,67 @@ function CstgAddDomainDialog({
   } = formMethods;
 
   const onSubmit = async (formData: AddDomainNamesFormProps) => {
-    const newDomainNames = separateStringsCommaSeparatedList(formData.newDomainNames);
-    // filter for uniqueness on what the user entered AND against existing domains
-    const dedupedDomains = deduplicateStrings(newDomainNames);
-    const uniqueDomains = dedupedDomains.filter((domain) => !existingDomains?.includes(domain));
+    const newDomains = separateStringsList(formData.newDomains);
+    // filter out domain names that already exist in the list
+    const uniqueDomains = newDomains.filter((domain) => !existingDomains?.includes(domain));
```

Comment:
Nit: They're not necessarily unique - the user might have entered the same domain twice.

Reply:
Yes - I am filtering again in line 66.  I originally had both filters in this line, but then I found a bug where two different valid domains entered changed to the same top-level domain, so I moved the second filter for deduplicating the list to after all domains are top-level.